File: Java/538.convert-bst-to-greater-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    int sum;
    public TreeNode convertBST(TreeNode root) {
        
        if(root == null){
            return root;
        }
        convertBST(root.right);
        int prev = root.val;
        root.val = sum + root.val;
        sum = sum + prev;
        convertBST(root.left);
        return root;            
    }
}

File: 538.convert-bst-to-greater-tree.js
/*
 * @lc app=leetcode id=538 lang=javascript
 *
 * [538] Convert BST to Greater Tree
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var convertBST = function (root) {
  if (root === null) return root;
  let curSum = 0;
  var dfs = function (node) {
    if (node === null) return;
    dfs(node.right);
    curSum += node.val;
    node.val = curSum;
    dfs(node.left);
  }
  dfs(root);
  return root;
};

